### ISSUE TITLE
Lock file not compatible with the current version of Poetry.

### ISSUE THREAD

**Issue by benjaminwoods** [benjaminwoods/derek#71]
From the following GitHub Actions job: (https://github.com/benjaminwoods/derek/actions/runs/7232234542/job/19706216196, #70)

```
Run poetry install
Creating virtualenv derek-py-s[7](https://github.com/benjaminwoods/derek/actions/runs/7232234542/job/19706216196?pr=70#step:6:8)Ijf0qn-py3.6 in /home/runner/.cache/pypoetry/virtualenvs

  RuntimeError

  The lock file is not compatible with the current version of Poetry.
  Upgrade Poetry to be able to read the lock file or, alternatively, regenerate the lock file with the `poetry lock` command.

  at /opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/poetry/packages/locker.py:4[8](https://github.com/benjaminwoods/derek/actions/runs/7232234542/job/19706216196?pr=70#step:6:9)2 in _get_lock_data
      478│                 "regenerate the lock file with the `poetry lock` command."
      47[9](https://github.com/benjaminwoods/derek/actions/runs/7232234542/job/19706216196?pr=70#step:6:10)│             )
      480│         elif not lock_version_allowed:
      481│             raise RuntimeError(
    → 482│                 "The lock file is not compatible with the current version of Poetry.\n"
      483│                 "Upgrade Poetry to be able to read the lock file or, alternatively, "
      484│                 "regenerate the lock file with the `poetry lock` command."
      485│             )
      486│ 
Error: Process completed with exit code 1.
```

**Comment by benjaminwoods** [comment]
Okay. The currnet lock file is generated by Poetry 1.7.1.

The problem here can be summarized as follows:
* This project supports Python >=3.6,<3.10.
* Poetry 1.7.1 is for Python >=3.8.

So, in the Python 3.6 job, we install a compatible version of Poetry, which is <1.7 (in this case, it appears to have been 1.1.15).

**Comment by benjaminwoods** [comment]
I don't like Option 3, because it shreds the lockfile every time. That's the same as just not having a lock file. This then screws with environments.

Option 1 is fine, but removes version support. Users in the Python community can be a bit slow to update their version of Python -- it's best to avoid removing version support where possible.

So, I think Option 2 is the best way to go. Recommended changes:
* update dev documentation to state that Poetry version needs to be 1.1.15 for this to work (therefore, probably best to use a local venv with poetry in it, a la `pipx`) -- do this separately
* rebuild lockfile using Poetry 1.1.15 + fix the version in CI (poetry<1.2).
